co2calculator/api/trip.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Trip classes"""
+
+
+from dataclasses import dataclass
+from pydantic import BaseModel
+
+from co2calculator.api.emission import Emissions
+from co2calculator.mobility.calculate_mobility import calc_co2_car
+from co2calculator.constants import CarFuel, Size, TransportationMode
+
+
+class Trip:
+    def __init__(self, distance=None, start=None, end=None):
+        """Initialize a trip object"""
+        self.__verify_parameters(distance, start, end)
+        self.distance = distance
+        self.start = start
+        self.end = end
+
+    def __verify_parameters(self, distance, start, end):
+        """Verifies whether the parameters passed by the user are valid"""
+        assert distance > 0, "Distance must be greater than 0"
+        if distance is None:
+            assert (
+                start is not None and end is not None
+            ), "If distance is given, start and end must be None."
+        elif distance is not None:
+            assert (
+                start is None and end is None
+            ), "If distance is not given, start and end must be given."
+
+    def by_car(self, fuel_type: str = None, size: str = None):
+        return _TripByCar(
+            fuel_type=fuel_type,
+            size=size,
+            distance=self.distance,
+            start=self.start,
+            end=self.end,
+        )
+
+
+class _TripByCar(Trip):
+    """This is a hidden class which handles car trips. It is not meant to be used by the user directly, instead it is only called from other classes of the module."""
+
+    transport_mode = TransportationMode.CAR
+
+    def __init__(
+        self,
+        fuel_type: str = None,
+        size: str = None,
+        distance=None,
+        start=None,
+        end=None,
+    ):
+        """Initialize a car trip"""
+        super(_TripByCar, self).__init__(distance=distance, start=start, end=end)
+        self.fuel_type = fuel_type
+        self.size = size
+
+    def calculate_co2e(self):
+        """
+        Calculate the CO2e emissions for a car trip
+
+        :param fuel_type: The fuel type of the car
+        :param size: The size of the car
+        :return: Emissions object
+        """
+
+        if self.distance is None:
+            distance = self.calculate_distance()
+
+        # Calculate emissions
+        options = {"fuel_type": self.fuel_type, "size": self.size}
```

Comment:
Wouldn't it make sense to use the `CarEmissionParameters` class here? Then we could easily evaluate the user input: 

```
options = CarEmissionParameters.parse_obj({"fuel_type": self.fuel_type, "size": self.size})
```
I think we would also save the next step where we check for empty values because Pydantic should set the appropriate default values.

Reply:
the problem is that `CarEmissionParameters` cannot take `None` values as input. So they have to be filtered out before. -_- It's really weird that it cannot handle that. Maybe we have to rethink using the pydantic classes here. But I would leave it for now until we work on issue #186

Reply:
Okay agree.